float64 prop1
float64 prop2
float64 prop3
float64 prop4 